src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx

```diff
@@ -264,28 +288,28 @@ const SatelliteManifestPanel: FunctionComponent<SatelliteManifestPanelProps> = (
     return rowsWithAllocationDetails;
   };
 
-  const toggleAllocationDetails = (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
-    const rowIndexToUpdate = rowKey / 2;
-    const newRowExpandedStatus = [...rowExpandedStatus];
-    newRowExpandedStatus[rowIndexToUpdate] = isOpen;
-    setRowExpandedStatus(newRowExpandedStatus);
+  const toggleAllocationDetails = (
+    event: React.MouseEvent,
+    rowKey: number,
+    isOpen: boolean,
+    rowData: any
+  ) => {
+    const uuid: string = rowData.uuid.title;
+    rowExpandedStatus[uuid] = !rowExpandedStatus[uuid];
+    forceUpdate();
   };
 
   const openCurrentEntitlementsListFromPanel = () => {
     closeDetailsPanel();
-
-    const newRowExpandedStatus = [...rowExpandedStatus];
-    newRowExpandedStatus[currentDetailRowIndex] = true;
-    setRowExpandedStatus(newRowExpandedStatus);
+    rowExpandedStatus[currentDetailUUID] = true;
```

Comment:
similar to the above, you shouldn't modify the piece of state directly.  Do something like: 

```
const newRowExpandedStatus = { ...rowExpandedStatus };
newRowExpandedStatus[currentDetailUUID] = true;
setRowExpandedStatus(newRowExpandedStatus);
```